scripts/research/lib/research_harness.py

```
@@ -0,0 +1,880 @@
+                f"outcome window end {end_ms} reaches/exceeds "
+                f"{self.policy.end_exclusive_ms}"
+            )
+
+
+def _run_git(
+    repo_root: Path,
+    *args: str,
+    input_bytes: bytes | None = None,
+) -> bytes:
+    try:
+        return subprocess.run(
```

Ruff (0.16.2)

[error] `subprocess` call: check for execution of untrusted input

(S603)